pkg/antctl/raw/test/command.go

```diff
@@ -0,0 +1,554 @@
+
+		_, err = k.clients.source.CreateDeployment(ctx, connectivityCheckNamespace, echoDeployment, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to create deployment %s: %s", echoSameNodeDeploymentName, err)
+		}
+
+		k.Log("[%s] Deploying client deployment...", k.clients.source.ClusterName())
+		clientDeployment := newDeployment(deploymentParameters{
+			Name:    clientDeploymentName,
+			Kind:    kindClientName,
+			Image:   "registry.k8s.io/e2e-test-images/agnhost:2.29",
+			Command: []string{"/agnhost", "pause"},
+			Port:    80,
+			Labels:  map[string]string{"app": clientDeploymentName},
+		})
+		_, err = k.clients.source.CreateDeployment(ctx, connectivityCheckNamespace, clientDeployment, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to create deployment %s: %s", clientDeploymentName, err)
+		}
+	}
+
+	if dstDeploymentNeeded {
+		k.Log("[%s] Deploying echo-other-node service...", k.clients.destination.ClusterName())
+		svc := newService(echoOtherNodeDeploymentName, map[string]string{"name": echoOtherNodeDeploymentName}, "http", 80)
+		_, err = k.clients.destination.CreateService(ctx, connectivityCheckNamespace, svc, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+
+		echoOtherNodeDeployment := newDeployment(deploymentParameters{
+			Name:    echoOtherNodeDeploymentName,
+			Kind:    kindEchoName,
+			Port:    80,
+			Image:   "k8s.gcr.io/e2e-test-images/agnhost:2.31",
+			Command: []string{"/agnhost", "netexec", "--http-port=80"},
+			Affinity: &corev1.Affinity{
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+						{
+							LabelSelector: &metav1.LabelSelector{
+								MatchExpressions: []metav1.LabelSelectorRequirement{
+									{Key: "name", Operator: metav1.LabelSelectorOpIn, Values: []string{clientDeploymentName}},
+								},
+							},
+							TopologyKey: "kubernetes.io/hostname",
+						},
+					},
+				},
+			},
+			Labels: map[string]string{"app": echoOtherNodeDeploymentName},
+		})
+
+		_, err = k.clients.destination.CreateDeployment(ctx, connectivityCheckNamespace, echoOtherNodeDeployment, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to create deployment %s: %s", echoOtherNodeDeploymentName, err)
+		}
+	}
+
+	return nil
+}
+
+func (k *k8sConnectivityParams) waitForDeploymentsReady(ctx context.Context, client k8sClientOperations, deployments []string) error {
+	k.Log("[%s] Waiting for deployments %s to become ready...", client.ClusterName(), deployments)
+
+	ctx, cancel := context.WithTimeout(ctx, k.params.podReadyTimeout())
+	defer cancel()
+	for _, name := range deployments {
+		for client.DeploymentIsReady(ctx, connectivityCheckNamespace, name) != nil {
+			select {
+			case <-time.After(time.Second):
+			case <-ctx.Done():
+				return fmt.Errorf("waiting for deployment %s to become ready has been interrupted: %w", name, ctx.Err())
+			}
+		}
+	}
```

Comment:
We usually leverage functions in `k8s.io/apimachinery/pkg/util/wait` for such things, in this case it would be `wait.PollUntilContextTimeout`

Reply:
Sure sir, implemented :)